### PR TITLE
35 implement customer management

### PR DESCRIPTION
**Pull request by ShalaevPavel**
Added customer controller, request to shopify app

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -51,7 +51,7 @@ gem 'bootsnap', require: false
 group :development, :test do
   # See https://guides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-debug-gem
   gem 'debug', platforms: %i[mri mingw x64_mingw]
-  gem 'pry'
+
```
Comment (fooooxmr) [inline Gemfile]: Why do you removing pry gem?

File: app/controllers/customers_controller.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  def new
+    @customer = Customer.new
+  end
+
+  def create
+    created_customer = create_customer_in_shopify_app
+    if created_customer
+      redirect_to root_path
+    else
+      # Handle failure case (e.g., show an error message to the user)
```
Comment (fooooxmr) [inline app/controllers/customers_controller.rb]: We should show error message to user in this case.

File: app/controllers/customers_controller.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  def update
+    customer_id = params[:id] # Assuming customer ID is passed as a parameter
+    updated_customer_data = params[:customer]
+
+    updated_customer = update_customer_in_shopify_app(customer_id, updated_customer_data)
+    if updated_customer
+      redirect_to root_path
+    else
+      # Handle failure case (e.g., show an error message to the user)
```
Comment (fooooxmr) [inline app/controllers/customers_controller.rb]: Show error somewhere.
When you are using GPT you need also review and update code a little bit :)

File: app/services/customer_creation.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
+require 'net/http'
+require 'json'
+
+def create_customer_in_shopify_app
```
Comment (fooooxmr) [inline app/services/customer_creation.rb]: It's a bad practice to create services like this.
Try to find another realisation of the services.
https://www.toptal.com/ruby-on-rails/rails-service-objects-tutorial
https://4ybakut2004.medium.com/%D0%BF%D0%B0%D1%82%D1%82%D0%B5%D1%80%D0%BD-service-object-%D0%B2-rails-11bd26a2efaf

**Review by fooooxmr — CHANGES_REQUESTED** [review]

File: app/controllers/products_controller.rb [hunk 5]
```diff
@@ -1,12 +1,16 @@
 class ProductsController < ApplicationController
-  before_action :set_product, only: %i[show edit update destroy]
-  before_action :create_session, only: %i[create update destroy]
+
```
Comment (fooooxmr) [inline app/controllers/products_controller.rb]: Why callbacks were removed?

File: app/controllers/customers_controller.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+  def edit
+    @customer = Customer.find(params[:id])
+  end
+
+  def update
+    customer_id = params[:id] # Assuming customer ID is passed as a parameter
+    updated_customer_data = params[:customer]
+
+    updated_customer = update_customer_in_shopify_app(customer_id, updated_customer_data)
```
Comment (fooooxmr) [inline app/controllers/customers_controller.rb]: Did you update customer data on the our side?

File: app/services/customer_creation.rb [hunk 7]
```diff
@@ -0,0 +1,58 @@
+require 'net/http'
```
Comment (evrusha) [inline app/services/customer_creation.rb]: For shopify requests, it's easier to use the shopify_api gem.

**Review by evrusha — CHANGES_REQUESTED** [review]

File: Gemfile.lock [hunk 8]
```diff
@@ -182,6 +180,12 @@ GEM
     regexp_parser (2.8.1)
     reline (0.3.5)
       io-console (~> 0.5)
+
+    securerandom (0.2.2)
+    shopify_api (13.1.0)
+=======
+=======
+>>>>>>> 8116b6ab5b4c1039a6dea8f1a3107c1e01fbfaf4
```
Comment (fooooxmr) [inline Gemfile.lock]: Fix it.

File: app/controllers/shopify_api_controller.rb [hunk 9]
```diff
@@ -1,4 +1,5 @@
-class ShopifyApiController < AuthorizedController
+class ShopifyApiController < ApplicationController
```
Comment (fooooxmr) [inline app/controllers/shopify_api_controller.rb]: AuthorizedController

File: app/models/adress.rb [hunk 10]
```diff
@@ -1,3 +1,10 @@
+<<<<<<< HEAD
+class Adress< ActiveRecord::Base
+  has_many :customers
+
+end
+=======
```
Comment (fooooxmr) [inline app/models/adress.rb]: Resolve conflicts.

Comment (fooooxmr) [reply app/models/adress.rb]: upper variant.

**Review by fooooxmr — APPROVED** [review]

**Review by Ikl0 — APPROVED** [review]

**Pull request merged**